project-ideas/order-ledger/createSalesOrder.md

```diff
@@ -1,47 +1,210 @@
    * Iterate through shipGroup.items as item
      * Set productSku = remove item.productSku
      * Call findOrCreate#Product for [internalName:productSku] map
      * Set item.productId = findOrCreateProductOutput.productId
-7. Set orderContext.shipGroups = orderJson.shipGroups
-8. Call create#org.apache.ofbiz.order.order.OrderHeader for orderContext
\ No newline at end of file
+7. Set payload.shipGroups = orderJson.shipGroups
+8. Call create#org.apache.ofbiz.order.order.OrderHeader for payload
+
+```json
+{
+    "currencyUom": "USD",
+    "customerClassificationId": null,
+    "entryDate": "2024-07-30T15:23:47+0000",
+    "externalId": "444455555",
+    "orderDate": "2024-07-25T04:07:49.956",
+    "orderName": "#499999",
+    "orderTypeId": "SALES_ORDER",
+    "statusId": "ORDER_CREATED",
+    "originFacilityId": "",
+    "presentmentCurrencyCode": "USD",
+    "priority": "2",
+    "productStoreId": "STORE",
+    "salesChannelEnumId": "UNKNWN_SALES_CHANNEL",
+    "webSiteId": "WEBSTORE",
+    "adjustments": [
+        {
+            "orderAdjustmentTypeId": "SHIPPING_CHARGES",
+            "amount": "500",
+            "exemptAmount": "2.5",
+            "comments": "",
+            "sourcePercentage": "5",
+            "adjustmentAttributes": [
+                {
+                    "attrName": "taxAuthority",
+                    "attrValue": "State of California",
+                    "attrDescription": "state tax"
+                }
+            ]
+        }
+    ],
+    "attributes": [
+        {
+            "attrName": "item",
+            "attrValue": "testing attribute",
+            "attrDescription": ""
+        }
+    ],
+    "email": "abc@gmail.com",
+    "customer": {
+        "partyId": "Doe",
+        "externalId": 22889483207024,
+        "first_name": "John",
+        "last_name": "Doe"
+    },
+    "billing_address": {
+        "email": "abc@gmail.com",
+        "first_name": "John",
+        "address1": "1020 Thalia Street",
+        "phone": "+17869573716",
+        "city": "New Orleans",
+        "zip": "70130",
+        "province": "Louisiana",
+        "country": "United States",
+        "last_name": "Doe",
+        "address2": null,
+        "company": null,
+        "latitude": 29.9377606,
+        "longitude": -90.07084449999999,
+        "name": "John Doe",
+        "country_code": "US",
+        "province_code": "LA"
+    },
+    "shipping_address": {
+        "email": "abc@gmail.com",
+        "first_name": "John",
+        "address1": "1020 Thalia Street",
+        "phone": "+17869573716",
+        "city": "New Orleans",
+        "zip": "70130",
+        "province": "Louisiana",
+        "country": "United States",
+        "last_name": "Doe",
+        "address2": null,
+        "company": null,
+        "latitude": 29.9377606,
+        "longitude": -90.07084449999999,
+        "name": "John Doe",
+        "country_code": "US",
+        "province_code": "LA"
```

Comment:
added billing_address map for OrderContactMechs SHIPPING_LOCATION, PHONE_SHIPPING and SHIPPING_EMAIL

Reply:
Yes, this should go in shipGroup level. I will update this.